### PR TITLE
Convert controller util and volume files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

**Comment by yangjunmyfm192085** [comment]
part duplicated with https://github.com/kubernetes/kubernetes/pull/100672

File: pkg/controller/volume/attachdetach/attach_detach_controller.go [hunk 1]
```diff
@@ -436,48 +436,41 @@ func (adc *attachDetachController) populateDesiredStateOfWorld() error {
...
 			plugin, err := adc.volumePluginMgr.FindAttachablePluginBySpec(volumeSpec)
 			if err != nil || plugin == nil {
-				klog.V(10).Infof(
-					"Skipping volume %q for pod %q/%q: it does not implement attacher interface. err=%v",
-					podVolume.Name,
-					podToAdd.Namespace,
-					podToAdd.Name,
-					err)
+				klog.V(10).InfoS(
+					"Skipping volume: it does not implement attacher interface.",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/attach_detach_controller.go:445]: Fixed.

File: pkg/controller/volume/attachdetach/attach_detach_controller.go [hunk 2]
```diff
@@ -727,21 +720,21 @@ func (adc *attachDetachController) processVolumeAttachments() error {
...
 		volumeName, err := volumeutil.GetUniqueVolumeNameFromSpec(plugin, volumeSpec)
 		if err != nil {
-			klog.Errorf(
-				"Failed to find unique name for volume:%q, va.Name:%q, nodeName:%q: %v",
-				*pvName,
-				va.Name,
-				nodeName,
-				err)
+			klog.ErrorS(
+				nil,
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/attach_detach_controller.go:724]: Fixed.

File: pkg/controller/volume/attachdetach/attach_detach_controller.go [hunk 3]
```diff
@@ -727,21 +720,21 @@ func (adc *attachDetachController) processVolumeAttachments() error {
...
 		attachState := adc.actualStateOfWorld.GetAttachState(volumeName, nodeName)
 		if attachState == cache.AttachStateDetached {
-			klog.V(1).Infof("Marking volume attachment as uncertain as volume:%q (%q) is not attached (%v)",
-				volumeName, nodeName, attachState)
+			klog.V(1).InfoS("Marking volume attachment as uncertain as volume is not attached",
+				"volume", volumeName, "node", nodeName, "attachState", attachState)
 			err = adc.actualStateOfWorld.MarkVolumeAsUncertain(volumeName, volumeSpec, nodeName)
 			if err != nil {
-				klog.Errorf("MarkVolumeAsUncertain fail to add the volume %q (%q) to ASW. err: %s", volumeName, nodeName, err)
+				klog.ErrorS(err, "MarkVolumeAsUncertain fail to add the volume to ASW.", "volume", volumeName, "node", nodeName)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/attach_detach_controller.go:737]: Fixed.

File: pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go [hunk 4]
```diff
@@ -143,15 +143,15 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 				informerPodUID := volutil.GetUniquePodName(informerPod)
 				// Check whether the unique identifier of the pod from dsw matches the one retrieved from pod informer
 				if informerPodUID == dswPodUID {
-					klog.V(10).Infof("Verified pod %q (UID %q) from dsw exists in pod informer.", dswPodKey, dswPodUID)
+					klog.V(10).InfoS("Verified Pod from dsw exists in pod informer.", "pod", dswPodKey, "podUID", dswPodUID)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go:146]: Fixed.

File: pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go [hunk 5]
```diff
@@ -143,15 +143,15 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
...
 					continue
 				}
 			}
 		}
 
 		// the pod from dsw does not exist in pod informer, or it does not match the unique identifier retrieved
 		// from the informer, delete it from dsw
-		klog.V(1).Infof("Removing pod %q (UID %q) from dsw because it does not exist in pod informer.", dswPodKey, dswPodUID)
+		klog.V(1).InfoS("Removing Pod from dsw because it does not exist in pod informer.", "pod", dswPodKey, "podUID", dswPodUID)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go:154]: Fixed.

File: pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go [hunk 6]
```diff
@@ -161,12 +161,12 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 		// The result is returned from CSIDriverLister which is from local cache. So this is not an expensive call.
 		volumeAttachable := volutil.IsAttachableVolume(volumeToAttach.VolumeSpec, dswp.volumePluginMgr)
 		if !volumeAttachable {
-			klog.Infof("Volume %v changes from attachable to non-attachable.", volumeToAttach.VolumeName)
+			klog.InfoS("Volume changes from attachable to non-attachable.", "volume", volumeToAttach.VolumeName)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go:164]: Fixed.

File: pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go [hunk 7]
```diff
@@ -161,12 +161,12 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
...
 			for _, scheduledPod := range volumeToAttach.ScheduledPods {
 				podUID := volutil.GetUniquePodName(scheduledPod)
 				dswp.desiredStateOfWorld.DeletePod(podUID, volumeToAttach.VolumeName, volumeToAttach.NodeName)
-				klog.V(4).Infof("Removing podUID: %v, volume: %v on node: %v from desired state of world"+
-					" because of the change of volume attachability.", podUID, volumeToAttach.VolumeName, volumeToAttach.NodeName)
+				klog.V(4).InfoS("Removing podUID, volume on Node from desired state of world"+
+					" because of the change of volume attachability.", "podUID", podUID, "volume", volumeToAttach.VolumeName, "node", volumeToAttach.NodeName)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go:169]: Fixed.

File: pkg/controller/volume/attachdetach/reconciler/reconciler.go [hunk 8]
```diff
@@ -110,11 +110,11 @@ func (rc *reconciler) reconciliationLoopFunc() func() {
 		rc.reconcile()
 
 		if rc.disableReconciliationSync {
-			klog.V(5).Info("Skipping reconciling attached volumes still attached since it is disabled via the command line.")
+			klog.V(5).InfoS("Skipping reconciling attached volumes still attached since it is disabled via the command line.")
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/reconciler/reconciler.go:113]: Fixed.

File: pkg/controller/volume/attachdetach/reconciler/reconciler.go [hunk 9]
```diff
@@ -110,11 +110,11 @@ func (rc *reconciler) reconciliationLoopFunc() func() {
...
 		} else if rc.syncDuration < time.Second {
-			klog.V(5).Info("Skipping reconciling attached volumes still attached since it is set to less than one second via the command line.")
+			klog.V(5).InfoS("Skipping reconciling attached volumes still attached since it is set to less than one second via the command line.")
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/reconciler/reconciler.go:115]: Fixed.

File: pkg/controller/volume/attachdetach/reconciler/reconciler.go [hunk 10]
```diff
@@ -172,51 +172,51 @@ func (rc *reconciler) reconcile() {
...
+			klog.V(5).InfoS(attachedVolume.GenerateMsgDetailed("Starting attacherDetacher.DetachVolume", ""))
 			verifySafeToDetach := !timeout
 			err = rc.attacherDetacher.DetachVolume(attachedVolume.AttachedVolume, verifySafeToDetach, rc.actualStateOfWorld)
 			if err == nil {
 				if !timeout {
-					klog.Infof(attachedVolume.GenerateMsgDetailed("attacherDetacher.DetachVolume started", ""))
+					klog.InfoS(attachedVolume.GenerateMsgDetailed("attacherDetacher.DetachVolume started", ""))
 				} else {
 					metrics.RecordForcedDetachMetric()
 					klog.Warningf(attachedVolume.GenerateMsgDetailed("attacherDetacher.DetachVolume started", fmt.Sprintf("This volume is not safe to detach, but maxWaitForUnmountDuration %v expired, force detaching", rc.maxWaitForUnmountDuration)))
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/reconciler/reconciler.go:222]: Fixed.

File: pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go [hunk 11]
```diff
@@ -65,15 +65,15 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
 		if errors.IsNotFound(err) {
 			// If node does not exist, its status cannot be updated.
 			// Do nothing so that there is no retry until node is created.
-			klog.V(2).Infof(
-				"Could not update node status. Failed to find node %q in NodeInformer cache. Error: '%v'",
-				nodeName,
-				err)
+			klog.V(2).InfoS(
+				"Could not update Node status. Failed to find Node in NodeInformer cache.",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go:69]: Fixed.

File: pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go [hunk 12]
```diff
@@ -65,15 +65,15 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
...
+				"node", nodeName,
+				"err", err)
 			continue
 		} else if err != nil {
 			// For all other errors, log error and reset flag statusUpdateNeeded
 			// back to true to indicate this node status needs to be updated again.
-			klog.V(2).Infof("Error retrieving nodes from node lister. Error: %v", err)
+			klog.V(2).InfoS("Error retrieving Nodes from Node lister.", "err", err)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go:76]: Fixed.

File: pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go [hunk 13]
```diff
@@ -83,10 +83,10 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
 			// to indicate this node status needs to be updated again
 			nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
 
-			klog.V(2).Infof(
-				"Could not update node status for %q; re-marking for update. %v",
-				nodeName,
-				err)
+			klog.V(2).InfoS(
+				"Could not update node status for Node; re-marking for update.",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go:87]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 14]
```diff
@@ -66,12 +66,12 @@ func CreateVolumeSpec(podVolume v1.Volume, pod *v1.Pod, nodeName types.NodeName,
 				err)
 		}
 
-		klog.V(10).Infof(
+		klog.V(10).InfoS(
 			"Found bound PV for PVC (ClaimName %q/%q pvcUID %v): pvName=%q",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:70]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 15]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
 	}
 
 	if len(pod.Spec.Volumes) <= 0 {
-		klog.V(10).Infof("Skipping processing of pod %q/%q: it has no volumes.",
-			pod.Namespace,
-			pod.Name)
+		klog.V(10).InfoS("Skipping processing of Pod: it has no volumes.",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:204]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 16]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
...
+			"podNamespace", pod.Namespace,
+			"pod", pod.Name)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:206]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 17]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
...
 	}
 
 	nodeName := types.NodeName(pod.Spec.NodeName)
 	if nodeName == "" {
-		klog.V(10).Infof(
-			"Skipping processing of pod %q/%q: it is not scheduled to a node.",
-			pod.Namespace,
-			pod.Name)
+		klog.V(10).InfoS(
+			"Skipping processing of Pod: it is not scheduled to a Node.",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:213]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 18]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
...
+			"podNamespace", pod.Namespace,
+			"pod", pod.Name)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:215]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 19]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
...
 		// If the node the pod is scheduled to does not exist in the desired
 		// state of the world data structure, that indicates the node is not
 		// yet managed by the controller. Therefore, ignore the pod.
-		klog.V(4).Infof(
-			"Skipping processing of pod %q/%q: it is scheduled to node %q which is not managed by the controller.",
-			pod.Namespace,
-			pod.Name,
-			nodeName)
+		klog.V(4).InfoS(
+			"Skipping processing of Pod: it is scheduled to Node which is not managed by the controller.",
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:222]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 20]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
...
-				"Error processing volume %q for pod %q/%q: %v",
-				podVolume.Name,
-				pod.Namespace,
-				pod.Name,
-				err)
+			klog.V(10).InfoS(
+				"Error processing volume for Pod",
+				"volume", podVolume.Name,
+				"podNamespace", pod.Namespace,
+				"pod", pod.Name,
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:237]: Fixed.

File: pkg/controller/volume/attachdetach/util/util.go [hunk 21]
```diff
@@ -201,53 +201,53 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
...
-				"Skipping volume %q for pod %q/%q: it does not implement attacher interface. err=%v",
-				podVolume.Name,
-				pod.Namespace,
-				pod.Name,
-				err)
+			klog.V(10).InfoS(
+				"Skipping volume for Pod: it does not implement attacher interface.",
+				"volume", podVolume.Name,
+				"podNamespace", pod.Namespace,
+				"pod", pod.Name,
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/attachdetach/util/util.go:249]: Fixed.

File: pkg/controller/volume/ephemeral/controller.go [hunk 22]
```diff
@@ -215,16 +215,16 @@ func (ec *ephemeralController) syncHandler(key string) error {
 	pod, err := ec.podLister.Pods(namespace).Get(name)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			klog.V(5).Infof("ephemeral: nothing to do for pod %s, it is gone", key)
+			klog.V(5).InfoS("Ephemeral: nothing to do for Pod, it is gone", "pod", key)
 			return nil
 		}
-		klog.V(5).Infof("Error getting pod %s/%s (uid: %q) from informer : %v", pod.Namespace, pod.Name, pod.UID, err)
+		klog.V(5).InfoS("Error getting Pod from informer", "podNamespace", pod.Namespace, "pod", pod.Name, "podUID", pod.UID, "err", err)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/ephemeral/controller.go:221]: Fixed.

File: pkg/controller/volume/expand/expand_controller.go [hunk 23]
```diff
@@ -219,19 +219,19 @@ func (expc *expandController) syncHandler(key string) error {
 		return nil
 	}
 	if err != nil {
-		klog.V(5).Infof("Error getting PVC %q (uid: %q) from informer : %v", util.GetPersistentVolumeClaimQualifiedName(pvc), pvc.UID, err)
+		klog.V(5).InfoS("Error getting PVC from informer", util.GetPersistentVolumeClaimQualifiedName(pvc), "pvcUID", pvc.UID, "err", err)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/expand/expand_controller.go:222]: Fixed.

File: pkg/controller/volume/expand/expand_controller.go [hunk 24]
```diff
@@ -286,7 +286,7 @@ func (expc *expandController) syncHandler(key string) error {
 			eventType = v1.EventTypeWarning
 		}
 		expc.recorder.Event(pvc, eventType, events.ExternalExpanding, fmt.Sprintf("Ignoring the PVC: %v.", msg))
-		klog.Infof("Ignoring the PVC %q (uid: %q) : %v.", util.GetPersistentVolumeClaimQualifiedName(pvc), pvc.UID, msg)
+		klog.InfoS("Ignoring the PVC.", "pvc", util.GetPersistentVolumeClaimQualifiedName(pvc), "pvcUID", pvc.UID, "err", msg)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/expand/expand_controller.go:289]: Fixed.

File: pkg/controller/volume/persistentvolume/framework_test.go [hunk 25]
```diff
@@ -801,7 +801,7 @@ func runMultisyncTests(t *testing.T, tests []controllerTest, storageClasses []*s
 				if err != nil {
 					if err == pvtesting.ErrVersionConflict {
 						// Ignore version errors
-						klog.V(4).Infof("test intentionally ignores version error.")
+						klog.V(4).InfoS("Test intentionally ignores version error.")
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/persistentvolume/framework_test.go:804]: Fixed.

File: pkg/controller/volume/persistentvolume/framework_test.go [hunk 26]
```diff
@@ -818,7 +818,7 @@ func runMultisyncTests(t *testing.T, tests []controllerTest, storageClasses []*s
...
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/persistentvolume/framework_test.go:821]: Fixed.

File: pkg/controller/volume/persistentvolume/framework_test.go [hunk 27]
```diff
@@ -953,7 +953,7 @@ func (plugin *mockVolumePlugin) Provision(selectedNode *v1.Node, allowedTopologi
 	}
 
 	plugin.provisionCallCounter++
-	klog.V(4).Infof("mock plugin Provision call nr. %d, returning %v: %v", plugin.provisionCallCounter, pv, call.ret)
+	klog.V(4).InfoS("Mock plugin Provision call nr.", "provisionCallCounter", plugin.provisionCallCounter, "pv", pv, "err", call.ret)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/persistentvolume/framework_test.go:956]: Fixed.

File: pkg/controller/volume/persistentvolume/framework_test.go [hunk 28]
```diff
@@ -975,7 +975,7 @@ func (plugin *mockVolumePlugin) Delete() error {
 	}
 	ret := plugin.deleteCalls[plugin.deleteCallCounter]
 	plugin.deleteCallCounter++
-	klog.V(4).Infof("mock plugin Delete call nr. %d, returning %v", plugin.deleteCallCounter, ret)
+	klog.V(4).InfoS("Mock plugin Delete call nr.", "deleteCallCounter", plugin.deleteCallCounter, "err", ret)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/persistentvolume/framework_test.go:978]: Fixed.

File: pkg/controller/volume/persistentvolume/framework_test.go [hunk 29]
```diff
@@ -1001,6 +1001,6 @@ func (plugin *mockVolumePlugin) Recycle(pvName string, spec *vol.Spec, eventReco
 	}
 	ret := plugin.recycleCalls[plugin.recycleCallCounter]
 	plugin.recycleCallCounter++
-	klog.V(4).Infof("mock plugin Recycle call nr. %d, returning %v", plugin.recycleCallCounter, ret)
+	klog.V(4).InfoS("Mock plugin Recycle call nr.", "recycleCallCounter", plugin.recycleCallCounter, "err", ret)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/volume/persistentvolume/framework_test.go:1004]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 I addressed all your comments.

**Pull request closed without merging**